src/gh-api-client.ts

```diff
@@ -30,85 +35,104 @@ type UpdateVariantAnalysis =
   | FailedAnalysis;
 
 export async function setVariantAnalysisRepoInProgress(
+  controllerRepoId: number,
   variantAnalysisId: number,
   repoId: number
 ): Promise<void> {
-  await updateVariantAnalysisStatus(variantAnalysisId, repoId, {
-    status: "in_progress",
-  });
+  await updateVariantAnalysisStatus(
+    controllerRepoId,
+    variantAnalysisId,
+    repoId,
+    {
+      status: "in_progress",
+    }
+  );
 }
 
 export async function setVariantAnalysisRepoSucceeded(
+  controllerRepoId: number,
   variantAnalysisId: number,
   repoId: number,
   sourceLocationPrefix: string,
   resultCount: number,
   databaseCommitSha: string
 ): Promise<void> {
-  await updateVariantAnalysisStatus(variantAnalysisId, repoId, {
-    status: "succeeded",
-    source_location_prefix: sourceLocationPrefix,
-    result_count: resultCount,
-    database_commit_sha: databaseCommitSha,
-  });
+  await updateVariantAnalysisStatus(
+    controllerRepoId,
+    variantAnalysisId,
+    repoId,
+    {
+      status: "succeeded",
+      source_location_prefix: sourceLocationPrefix,
+      result_count: resultCount,
+      database_commit_sha: databaseCommitSha,
+    }
+  );
 }
 
 export async function setVariantAnalysisFailed(
+  controllerRepoId: number,
   variantAnalysisId: number,
   repoId: number,
   failureMessage: string
 ): Promise<void> {
-  await updateVariantAnalysisStatus(variantAnalysisId, repoId, {
-    status: "failed",
-    failure_message: failureMessage,
-  });
+  await updateVariantAnalysisStatus(
+    controllerRepoId,
+    variantAnalysisId,
+    repoId,
+    {
+      status: "failed",
+      failure_message: failureMessage,
+    }
+  );
 }
 
 async function updateVariantAnalysisStatus(
+  controllerRepoId: number,
   variantAnalysisId: number,
   repoId: number,
   data: UpdateVariantAnalysis
 ): Promise<void> {
-  const http = getApiClient();
+  const octokit = getOctokit();
 
-  const url = `/codeql/variant-analyses/${variantAnalysisId}/repositories/${repoId}`;
-  const response = await http.patch(url, JSON.stringify(data));
-  if (response.message.statusCode !== 204) {
+  const url = `PATCH /repositories/${controllerRepoId}/code-scanning/codeql/variant-analyses/${variantAnalysisId}/repositories/${repoId}`;
+  const response = await octokit.request(url, { data });
```

Comment:
Octokit throws exceptions for failure requests e.g. 404 so we'll need to have try/catch around `octokit.request` to capture those. Unfortunately simply checking the success code is not enough.

Reply:
Alternatively, do we need to capture the exceptions or just let them bubble up? If the exception is descriptive enough already to be useful in debugging then it's fine and we could remove all the status-checking code entirely.

Reply:
I've removed the status-checking entirely, so hopefully octokit's errors will be descriptive enough.

(We can't really test until I've rebased https://github.com/github/codeql-variant-analysis-action/pull/740 onto this though 🙃 )